### ISSUE TITLE
ValueError: filedescriptor out of range in select()

### ISSUE THREAD

**Issue by ipv6freely** [ncclient/ncclient#254]
I'm using [nornir](https://github.com/nornir-automation/nornir) to automate configuration of about 400 devices in parallel. 

I'm getting this traceback:
```
Traceback (most recent call last):
  File "/Users/cjones/.local/share/virtualenvs/python-LG6-DRlH/lib/python3.6/site-packages/jnpr/junos/device.py", line 1251, in open
    device_params={'name': 'junos', 'local': False})
  File "/Users/cjones/.local/share/virtualenvs/python-LG6-DRlH/lib/python3.6/site-packages/ncclient/manager.py", line 162, in connect
    return connect_ssh(*args, **kwds)
  File "/Users/cjones/.local/share/virtualenvs/python-LG6-DRlH/lib/python3.6/site-packages/ncclient/manager.py", line 127, in connect_ssh
    session.connect(*args, **kwds)
  File "/Users/cjones/.local/share/virtualenvs/python-LG6-DRlH/lib/python3.6/site-packages/ncclient/transport/ssh.py", line 451, in connect
    self._post_connect()
  File "/Users/cjones/.local/share/virtualenvs/python-LG6-DRlH/lib/python3.6/site-packages/ncclient/transport/session.py", line 107, in _post_connect
    raise error[0]
  File "/Users/cjones/.local/share/virtualenvs/python-LG6-DRlH/lib/python3.6/site-packages/ncclient/transport/ssh.py", line 539, in run
    r, w, e = select([chan], [], [], TICK)
ValueError: filedescriptor out of range in select()
```
A lot of googling has brought me to the same conclusion - that select() should be replaced with kqueue. Have a look at these:

https://github.com/ansible/ansible/issues/10157
https://github.com/ansible/ansible/pull/20934

I'm on macOS 10.14 using python 3.6. I have already tried increasing the open files limit on my system to 200,000 but that hasn't helped.

**Comment by ipv6freely** [comment]
Fantastic. Will try and report back.

**Comment by einarnn** [comment]
Great, thanks!  I meant to ask explicitly if you could test in your scaled environment as I don't have one currently.

Cheers,

Einar

**Comment by ipv6freely** [comment]
I assumed that's what you meant :)

**Comment by einarnn** [comment]
Due to a mistaken commit, these changes are now in PR #260. Sorry for any inconvenience.

**Comment by einarnn** [comment]
Great! Will see if I can get that committed and spin a new release.

**Comment by einarnn** [comment]
Addressed in #260.

**Comment by einarnn** [comment]
Re
